shared/mtemplates/templates.go

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package mtemplates
+
+import (
+	"bytes"
+	"html/template"
+	"io"
+	"path/filepath"
+	"sync/atomic"
+
+	"github.com/fsnotify/fsnotify"
+
+	"github.com/mattermost/mattermost-server/v5/mlog"
+)
+
+type Templates struct {
```

Comment:
Is this a generic template or it is an HTML template? Since from reading the struct name i assume that is a generic template, but reading the test cases i see that we are testing an HTML template. Maybe worth changing the struct name to HTMLTemplate, also worth dropping the plural i guess.

Reply:
Actually, this is the "Templates" bucket, where you initialize the templates, and then, using that templates you render things. So the plurar makes sense. I think adding the `HTML` prefix can be more explicit.

Reply:
+1 on ☝️

Reply:
Done